### PR TITLE
support query cargo config value

### PR DESCRIPTION
**Pull request by LuoZijun**
In one of my own projects, I wrote a dynamic link library for Golang using Rust. When compiling Golang code, I need to know from which directory to copy the compiled dynamic link library.

`target-dir` is not always in` $PROJECT-ROOT/target`. Because some people have a global `$HOME/.cargo/config`, and some projects have their own` $PROJECT-ROOT/.cargo/config`.

So I need to read the `.cargo/config` file to see if`build.target-dir` is set in it. So that I know where to copy the compiled dynamic link library.

I found that your project only handles `manifest-file`, so I added processing code for` $PROJECT-ROOT/.cargo/config` and `$HOME/.cargo/config` to your project .

After writing these, I found out that I don't actually need to do this myself. This can be done using the official command (`cargo metadata --format-version 1 | jq -r ".target_directory "`).

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1,21 +0,0 @@
-[root]
```
Comment (wesleywiser) [inline Cargo.lock:1]: `Cargo.lock` is generally supposed to be checked into git for executables. Why do you need to delete it?

https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

File: .gitignore [hunk 2]
```diff
@@ -1 +1,5 @@
-target
+.DS_STORE
+
+/target
+**/*.rs.bk
+Cargo.lock
```
Comment (wesleywiser) [inline .gitignore]: `Cargo.lock` should not be ignored.

**Review by wesleywiser — COMMENTED** [review]
Thanks for the PR! 

Since there are a lot of changes to the code, could you summarize what this PR does?

File: Cargo.lock (hunk 1 above)
Comment (LuoZijun) [reply Cargo.lock:1]: This is my personal habit, because I think `cargo` will generate this file by itself, so there is no need to put it in the source code for tracking.
I respect different habits, so I will submit this file later.

**Review by wesleywiser — APPROVED** [review]

**Comment by wesleywiser** [comment]
Gotcha, thanks!
